src/sidebar/services/api.js

```diff
@@ -126,8 +139,15 @@ function createAPICall(
           descriptor.url,
           params
         );
+
         const apiUrl = new URL(url);
-        apiUrl.search = queryString.stringify(queryParams);
+        for (let [key, value] of Object.entries(queryParams)) {
+          if (Array.isArray(value)) {
+            value.forEach(v => apiUrl.searchParams.append(key, v.toString()));
+          } else {
+            apiUrl.searchParams.append(key, value.toString());
+          }
+        }
```

Comment:
May I beg for some commenting here? I started looking at this module, and tracking back to `util/url` to grok what `replaceURLParams` is all about. I don't immediately understand what "unused parameters" means, why it's getting renamed to `queryParams` here and what we're doing with it. Some commenting context would be very helpful! I can work through what it's doing here (handling Array parameter values differently than non-Arrays), but comments for the next guy would be a gift.